### PR TITLE
Use native Lucene locks for NFS safety

### PR DESCRIPTION
**Pull request by seanf**
Partly reverts https://github.com/zanata/zanata-server/pull/562

### CODE REVIEW

File: zanata-war/src/main/webapp-jboss/WEB-INF/classes/META-INF/persistence.xml [hunk 1]
```diff
@@ -87,6 +87,8 @@
       <!-- Search engine -->
       <property name="hibernate.search.default.directory_provider"
         value="org.hibernate.search.store.impl.FSDirectoryProvider" />
+      <property name="hibernate.search.default.locking_strategy"
+        value="simple" />
```
Comment (carlosmunoz) [inline zanata-war/src/main/webapp-jboss/WEB-INF/classes/META-INF/persistence.xml]: This is the default for FSDirectoryProvider correct? Are you just being explicit or is there something else?

**Comment by carlosmunoz** [comment]
Other than my question, looks good.
:+1:

File: zanata-war/src/main/webapp-jboss/WEB-INF/classes/META-INF/persistence.xml (hunk 1 above)
Comment (seanf) [inline zanata-war/src/main/webapp-jboss/WEB-INF/classes/META-INF/persistence.xml]: Apparently, as of Hibernate Search 4.1, the default is native.

And going by our production logs, we have been using NativeFSLock:

> (Hibernate Search: Index updates queue processor for index org.zanata.model.HTextFlowTarget-1) [] HSEARCH000058: HSEARCH000117: IOException on the IndexWriter: org.apache.lucene.store.LockReleaseFailedException: Cannot forcefully unlock a NativeFSLock which is held by another indexer component: /var/lib/zanata/index/org.zanata.model.HTextFlowTarget/write.lock

**Comment by djansen-redhat** [comment]
@seanf Not sure how this will help me in Need For Speed.  How do I test?

**Comment by seanf** [comment]
Ideally, by using a filesystem running NFS.  But I would suggest killing
Zanata (abnormally) to leave some .lock files in the index directory, and
then restart Zanata - the deployment should fail, with a reasonably helpful
error message about the .lock files.

On 27 October 2015 at 14:55, Damian Jansen <notifications@github.com> wrote:

> @seanf <https://github.com/seanf> Not sure how this will help me in Need
> For Speed. How do I test?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/1021#issuecomment-151375833>
> .
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat

**Comment by djansen-redhat** [comment]
:white_check_mark: Tested

**Pull request merged**